addons/point_of_sale/static/src/app/store/models.js

```diff
@@ -1431,7 +1431,10 @@ export class Order extends PosModel {
         });
         paymentLines = [];
         this.paymentlines.forEach((item) => {
-            return paymentLines.push([0, 0, item.export_as_JSON()]);
+            const itemAsJson = item.export_as_JSON();
+            if (itemAsJson) {
+                return paymentLines.push([0, 0, itemAsJson]);
+            }
```

Comment:
Can you try again without this diff? You might have encountered this because of how the reactivity works. When removing items in a collection (array), the item is removed but the length is not updated, so there is a time where the an `undefined` slot is present in the array. This can be avoided if all the reactive effects we have is "batched".

Reply:
No this is because I return null for online payment lines here: https://github.com/odoo/odoo/pull/123237/files/0f62f4170488cdfc1439ef49452d971c3386ec93#diff-bdcc8aaeaf691939f64fe9da597375bb5145f498c84ecbc487766d489e11fde0R94-R111
This working principle can seem strange, but it is the easiest way, in my opinion, of never sending online payment lines to the server in the current state of the code. Changing this would require to change a lot of things in the way orders are sent to the server.

Reply:
Oh. Understood.